### ISSUE TITLE
Definition of zoom level does not follow conventions

### ISSUE THREAD

**Issue by mainkai** [msteiger/jxmapviewer2#87]
Hi,
first of all, thanks for the great library.

I noticed that the implementation of zoom levels uses a strange and unclear [definition](https://github.com/msteiger/jxmapviewer2/blob/c6fefa78eb68fe8e7d5bc318863d740a9b007afa/jxmapviewer2/src/main/java/org/jxmapviewer/JXMapViewer.java#L67), leading to 0 being the maximum possible zoom level and requiring calculations with hardcoded values like [here](https://github.com/msteiger/jxmapviewer2/blob/c6fefa78eb68fe8e7d5bc318863d740a9b007afa/jxmapviewer2/src/main/java/org/jxmapviewer/viewer/wms/WMSTileFactory.java#L32).

Wouldn't it be better to just use [the definition of zoom levels that OpenStreetMap uses](https://wiki.openstreetmap.org/wiki/Zoom_levels), i.e. zoom 0 = whole world with 360° Tile width and an unconstrained number of higher zoom levels?

Maybe that could also help with [that issue](https://github.com/msteiger/jxmapviewer2/issues/74), since zoom level 0 would not be the maximum any more, but the minimum instead. So zoom levels > 17 should not be a problem any more...

**Comment by msteiger** [comment]
Yeah, I agree that it is odd. I believe that the super-old Blue Marble renderer worked that way and the original authors made OSM behave the same. Changing it now is difficult, since it would break the compatibility with all existing implementations, so I left it was it was. Not sure if anything can be done here with reasonable effort.

**Comment by Heiko-Zelt** [comment]
I wondered too because I used the Leaflet JavaScript library before and zoom levels are reversed.
